std/file.d

```diff
@@ -5274,7 +5274,21 @@ Returns:
 */
 string tempDir() @trusted
 {
-    import std.path : dirSeparator;
+    // We must check that the end of a path is not a seperator, before adding another
+    // If we don't we end up with https://issues.dlang.org/show_bug.cgi?id=22738
+    static string addSeperator(string input)
+    {
+        import std.path : dirSeparator;
+        import std.algorithm.searching : endsWith;
+        
+        // It is very rare a directory path will reach this point with a directory seperator at the end
+        // However on OSX this can happen, so we must verify least we break user code i.e. https://github.com/dlang/dub/pull/2208
```

Comment:
Typo (https://brians.wsu.edu/2016/05/19/least-lest/)
```suggestion
        // However on OSX this can happen, so we must verify lest we break user code i.e. https://github.com/dlang/dub/pull/2208
```